staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/draplugin.go

```diff
@@ -294,22 +392,25 @@ func Start(ctx context.Context, nodeServers []interface{}, opts ...Option) (resu
 	if o.driverName == "" {
 		return nil, errors.New("driver name must be set")
 	}
-	if o.draAddress == "" {
-		return nil, errors.New("DRA address must be set")
+	if o.pluginRegistrationEndpoint.file == "" {
+		o.pluginRegistrationEndpoint.file = o.driverName + "-reg.sock"
 	}
-	var emptyEndpoint endpoint
-	if o.draEndpoint == emptyEndpoint {
-		return nil, errors.New("a Unix domain socket path and/or listener must be set for the kubelet plugin")
+	if o.pluginDataDirectoryPath == "" {
+		o.pluginDataDirectoryPath = path.Join(KubeletPluginsDir, o.driverName)
 	}
-	if o.pluginRegistrationEndpoint == emptyEndpoint {
-		return nil, errors.New("a Unix domain socket path and/or listener must be set for the registrar")
+	draEndpoint := endpoint{
```

Comment:
nit: Since we're moving from defaults to actual initialization, should this be moved further down? Also, does it make sense to instantiate `draEndpoint` just before we need it? Would this make factoring code out later easier?

Reply:
Moved.